maturity-level-two/docs/content-negotiation.md

```diff
@@ -0,0 +1,52 @@
+# Content Negotiation
+
+ When no specific compatibility requirements regarding the rest request and response formats are set, it is recommended to use the JSON format (application/json). But in some situations a client might require the data to be formatted in a certain format in order to interpret the data correctly. With content negotiation we determine in what format we'd like to receive requests  & responses. For REST API's, the most common formats are JSON and XML.
+As an API consumer, you can specify what format you are expecting by adding HTTP headers:
+- `Content-Type` - Specify the format of the request
+- `Accept` - Specify the requested format of the response. Default format will be used when not specified.
+When a format is not supported, the API should return an [HTTP 406 Not Acceptable](https://httpstatuses.com/406).
+Because of it's lightness and fastness, JSON has become the standard over the last couple of years but in certain situations other formats still have their advantages. In addition to this, ASP.NET Core also uses some additional formatters for special cases, such as the TextOutputFormatter and the HttpNoContentOutputFormatter.
+
+ When you would like to make sure your api only uses the JSON format, you can specify this in the startup of your ASP.NET Core project. This will make sure you'll refuse all non-JSON 'Accept' headers, including the plain text headers (on these requests you will return response code 406). If no 'Accept' header is specified you can return JSON as it is the only supported type. You can find an example of the changes you have to do to the startup below: 
+ ```csharp
+services.AddMvc(options =>
+{
+    var jsonInputFormatters = options.InputFormatters.OfType<JsonInputFormatter>();
+    var jsonInputFormatter = jsonInputFormatters.First();
+    options.InputFormatters.Clear();
+    options.InputFormatters.Add(jsonInputFormatter);
+    var jsonOutputFormatters = options.OutputFormatters.OfType<JsonOutputFormatter>();
+    var jsonOutputFormatter = jsonOutputFormatters.First();
+    options.OutputFormatters.Clear();
+    options.OutputFormatters.Add(jsonOutputFormatter);
+}).SetCompatibilityVersion(CompatibilityVersion.Version_2_2);
+```
+In here we have to add the 'SetCompatibilityVersion'as well to make sure the supported formats are documented correctly in e.g. the swagger. In case you'd like to add other formatting possibilities, it is possible to add these formatters to your api formatters. In case of xml you can use the XmlSerializerInputFormatter and the XmlSerializerOutputFormatter or add the xml formatters using the Mvc. With this approach the default format is still JSON.
```

Comment:
One question : just imagine that i want to return XML for only a single operation. What should i do?
A better scenario is : I have an API with multiple operations, one of those is a GET that returns a list of items.
I want to be able to get the list of items in "text/csv" what should be the approach?

Reply:
You can uses the [Produces("text/csv") but this will not allow the "text/csv" as an acceptable "accept' header..